MiniBootcamp/Modules/Feed/FeedViewController.swift

```diff
@@ -6,29 +6,147 @@
+        navigationController?.navigationBar.isTranslucent = false
+    }
+    
+    private func configureMenuButton(){
+        // Initialization
+        let menuBtn = UIButton(type: .custom)
+        menuBtn.frame = CGRect(x: 0.0, y: 0.0, width: 20, height: 20)
+        menuBtn.setImage(UIImage(named:"plus-2"), for: .normal)
+//        menuBtn.addTarget(self, action: #selector(addTapped), for: UIControl.Event.touchUpInside)
+        //Configuration
+        let menuBarItem = UIBarButtonItem(customView: menuBtn)
+        let currWidth = menuBarItem.customView?.widthAnchor.constraint(equalToConstant: 24)
+        currWidth?.isActive = true
+        let currHeight = menuBarItem.customView?.heightAnchor.constraint(equalToConstant: 24)
+        currHeight?.isActive = true
+        self.navigationItem.rightBarButtonItem = menuBarItem
+    }
+    
+    
+    private func configureTableView() {
+        
+        
         tableView.dataSource = self
         tableView.delegate = self
-        view.addSubview(tableView)
+        tableView.register(TweetTableViewCell.self, forCellReuseIdentifier: TweetTableViewCell.identifier)
+//        tableView.rowHeight = UITableView.automaticDimension
+//        tableView.estimatedRowHeight = 44
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        self.view.addSubview(tableView)
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.topAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+    }
+    
+    private func configureActivityIndicatorView() {
         
-        tableView.anchor(top: view.topAnchor, leading: view.leadingAnchor, trailing: view.trailingAnchor, bottom: view.bottomAnchor)
+        self.view.addSubview(activityIndicator)
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            activityIndicator.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
+        activityIndicator.startAnimating()
+    }
+    
+    func fetchTweets() {
+        TweetTimelineAPI().load(.timeline) { result in
+            switch(result) {
+            case .success(let _tweets):
+                self.tweetsViewModel = TweetListViewModel(_tweets)
+                DispatchQueue.main.async {
+                    self.waitDownload()
+                }
+            case .failure(_):
+                print("Couldn't fetch")
+            }
+        }
+    }
+    private func waitDownload() {
+        
+        for tweet in self.tweetsViewModel!.getListTweets() {
+            let url = URL(string: tweet.user.profileImageURL!)!
+            self.group.enter()
```

Comment:
Declare the dispatch group locally to prevent double calls if this method is not sync with the user flow.
Alternatively You can use NSOperationQueue for this class property management of sync operations